classes/plugins/ThemePlugin.inc.php

```diff
@@ -28,107 +51,358 @@ function ThemePlugin() {
+		return $activeTheme == basename($this->getPluginPath());
+	}
 
-				// If this theme uses templates, ensure they're given priority.
-				array_unshift(
-					$templateManager->template_dir,
-					$path = Core::getBaseDir() . DIRECTORY_SEPARATOR . $this->getPluginPath() . '/templates'
-				);
+	/**
+	 * Add a stylesheet to load with this theme
+	 *
+	 * Style paths with a .less extension will be compiled and redirected to
+	 * the compiled file.
+	 *
+	 * @param string $name A name for this stylesheet
+	 * @param string $path The path to this stylesheet, relative to the theme
+	 * @param array $args Optional arguments hash. Supported args:
+	 *   'context': Whether to load this on the `frontend` or `backend`.
+	 *      default: `frontend`
+	 *   'priority': Controls order in which styles are printed
+	 *   'addLess': Additional LESS files to process before compiling. Array
+	 */
+	public function addStyle($name, $path, $args = array()) {
+
+		// Pass a file path for LESS files
+		if (substr($path, -4) == 'less' ) {
+			$fullPath = $this->_getBaseDir($path);
+
+		// Pass a URL for other files
+		} else {
+			if (isset($args['baseUrl'])) {
+				$fullPath = $args['baseUrl'] . $path;
+			} else {
+				$fullPath = $this->_getBaseUrl($path);
 			}
 		}
-		return true;
+
+		// Generate file paths for any additional LESS files to compile with
+		// this style
+		if (isset($args['addLess'])) {
+			foreach ($args['addLess'] as &$file) {
+				$file = $this->_getBaseDir($file);
+			}
+		}
+
+		$this->styles[$name] = array(
+			'path' => $fullPath,
+			'context' => isset($args['context']) && $args['context'] == 'backend' ? 'backend' : 'frontend',
+			'priority' => isset($args['priority']) ? (int) $args['priority'] : STYLE_SEQUENCE_NORMAL,
+			'addLess' => isset($args['addLess']) ? (array) $args['addLess'] : array(),
+			'baseUrl' => isset($args['baseUrl']) ? (string) $args['baseUrl'] : '',
+		);
+	}
+
+	/**
+	 * Modify the params of an existing stylesheet
+	 *
+	 * @param string $name The name of the stylesheet to modify
+	 * @param array $args Parameters to modify.
+	 * @see self::addStyle()
+	 * @return null
+	 */
+	public function modifyStyle($name, $args = array()) {
+
+		$style = &$this->getStyle($name);
+
+		if (empty($style)) {
+			return;
+		}
+
+		if (isset($args['addLess'])) {
+			foreach ($args['addLess'] as &$file) {
+				$file = $this->_getBaseDir($file);
+			}
+		}
+
+		if (isset($args['path'])) {
+			$args['path'] = substr($args['path'], -4) == 'less' ? $this->_getBaseDir($path) : $this->_getBaseUrl($args['path']);
+		}
+
+
+		$style = array_merge( $style, $args );
+	}
+
+	/**
+	 * Get a style from this theme or any parent theme
+	 *
+	 * @param string $name The name of the style to retrieve
+	 * @return array|null Reference to the style or null if not found
+	 */
+	public function &getStyle($name) {
```

Comment:
I believe so, though I confess references aren't exactly my forte. The function is used as a helper function to retrieve any style by it's name, whether that style is in the current theme or a parent theme (we actually support unlimited parent/child nesting, and `getStyle()` will go up the chain recursively):

https://github.com/pkp/pkp-lib/pull/1519/files#diff-a8bc8a63802d1659d8ddc41a44c55b9fR162

The reference makes it easy to retrieve a style and modify it without having to know it's exact position in the parent/child theme hierarchy.